pkg/util/ip_address.go

```diff
@@ -38,12 +38,12 @@ type NetworkAddress struct {
 }
 
 // SplitHostPortDefault splits ip address/hostname string by host and port. Defaults used if no match found
-func SplitHostPortDefault(input, defaultHost, defaultPort string) (NetworkAddress, error) {
+func SplitHostPortDefault(input, defaultHost, defaultPort string, allowEmpty bool) (NetworkAddress, error) {
 	addr := NetworkAddress{
 		Host: defaultHost,
 		Port: defaultPort,
 	}
-	if len(input) == 0 {
+	if len(input) == 0 && !allowEmpty {
```

Comment:
To me the correct handling if `input` is empty would be to return `addr`. What do you think?